include/IDriver.h

```diff
@@ -133,11 +132,7 @@ namespace video
 
             virtual core::smart_refctd_ptr<IGPURenderpassIndependentPipeline> createGPURenderpassIndependentPipeline(
                 core::smart_refctd_ptr<IGPUPipelineLayout>&& _layout,
-                core::smart_refctd_ptr<IGPUSpecializedShader>&& _vs,
-                core::smart_refctd_ptr<IGPUSpecializedShader>&& _tcs,
-                core::smart_refctd_ptr<IGPUSpecializedShader>&& _tes,
-                core::smart_refctd_ptr<IGPUSpecializedShader>&& _gs,
-                core::smart_refctd_ptr<IGPUSpecializedShader>&& _fs,
+                IGPUSpecializedShader** _shaders,
```

Comment:
iterator pair?

Reply:
look at docs

Reply:
I understand why its an array of 5, but thats not future proof.

Reply:
why not? all you have to do is to adjust value of SHADER_STAGE_COUNT.

Reply:
Because its absolutely pointless to shove a full array of 14 elements down this pointer when you're only going to use 6 maximum, also its:
A. Unsized (when I look at code suggestions in the IDE its not apparent how many elements this function needs)
B. putting the shaders in wrong order will fuck everything up, despite the fact that `IGPUSpecializedShader` knows its own stage and entry point.

Reply:
resolved

Reply:
https://github.com/buildaworldnet/IrrlichtBAW/pull/358#discussion_r332878486